pkg/controllers/user/logging/utils/daemonset.go

```diff
@@ -51,28 +50,48 @@ func CreateFluentd(ds rv1beta2.DaemonSetInterface, sa rv1.ServiceAccountInterfac
 	return nil
 }
 
-func RemoveFluentd(ds rv1beta2.DaemonSetInterface, sa rv1.ServiceAccountInterface, rb rrbacv1.ClusterRoleBindingInterface) error {
-	deleteOp := metav1.DeletePropagationBackground
-	var errgrp errgroup.Group
-	errgrp.Go(func() error {
-		return ds.Delete(loggingconfig.FluentdName, &metav1.DeleteOptions{PropagationPolicy: &deleteOp})
-	})
-
-	errgrp.Go(func() error {
-		return sa.Delete(loggingconfig.FluentdName, &metav1.DeleteOptions{})
-	})
+func CreateLogAggregator(ds rv1beta2.DaemonSetInterface, sa rv1.ServiceAccountInterface, rb rrbacv1.ClusterRoleBindingInterface, cl v3.ClusterLister, clusterName, namespace string) (err error) {
+	serviceAccount := newServiceAccount(loggingconfig.LogAggregator, namespace)
+	serviceAccount, err = sa.Create(serviceAccount)
+	defer func() {
+		if err != nil {
+			sa.Delete(serviceAccount.Name, &metav1.DeleteOptions{})
```

Comment:
why are we doing the cleanup? What if the error == alreadyExists error? If we need to have a deferred delete, why are we having 3 anonymous functions instead of 1? 

I'm also not very comfortable that we don't care about the return value of delete function. It means that instead of returning the error to the framework - which will end up in framework rescheduling the job - we should handle the failed cleanup in a custom manner. What if there are "leftovers" from removal failure? Do we clean them up some place else?